### PR TITLE
Interpolate historical data too

### PR DESCRIPTION
**Pull request by potash**
For consistency it makes sense to interpolate the historical data points too

I also made a slight refactor to `write_tweet()` so that it only takes `end_time` and `timespan` (removing the redundant parameter `start_time`). Makes it easier to test and less error-prone.

### CODE REVIEW

**Comment by beamalsky** [comment]
Just so I know: Is there anywhere in your historical data that it's not compiled hourly? Ie, are there any gaps?

File: tweets.py [hunk 1]
```diff
@@ -106,16 +107,26 @@ def get_historical_temps(place, start_time, end_time):
...
     try:
         observed_temp = get_observed_temp(place, start_time, end_time)
     except ValueError as e:
         capture_exception(e)
         return
 
-    historical_temps = get_historical_temps(place, start_time, end_time)
+    historical_temps = get_historical_temps(place, start_time, end_time).set_index('timestamp')
+
+    def average_interp_named_timeseries(timeseries):
```
Comment (beamalsky) [inline tweets.py]: Did you mean to put this under `write_tweet`? I get this error: 

```
Beas-MacBook-Pro:itww_bot beamalsky$ docker-compose run --rm app python bot.py
INFO:root:Coverage: 0 days 01:00:00
Traceback (most recent call last):
  File "bot.py", line 32, in <module>
    print(get_tweets(place))
  File "/app/tweets.py", line 31, in get_tweets
    tweet = write_tweet(place, end_time, DAY)
  File "/app/tweets.py", line 129, in write_tweet
    averages = historical_temps.groupby('interval', observed=True).temp.agg(average_interp_named_timeseries)
  File "/usr/local/lib/python3.7/site-packages/pandas/core/groupby/generic.py", line 265, in aggregate
    return self._python_agg_general(func, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/pandas/core/groupby/groupby.py", line 936, in _python_agg_general
    result, counts = self.grouper.agg_series(obj, f)
  File "/usr/local/lib/python3.7/site-packages/pandas/core/groupby/ops.py", line 641, in agg_series
    return self._aggregate_series_fast(obj, func)
  File "/usr/local/lib/python3.7/site-packages/pandas/core/groupby/ops.py", line 666, in _aggregate_series_fast
    result, counts = grouper.get_result()
  File "pandas/_libs/reduction.pyx", line 376, in pandas._libs.reduction.SeriesGrouper.get_result
  File "pandas/_libs/reduction.pyx", line 193, in pandas._libs.reduction._BaseGrouper._apply_to_group
  File "/usr/local/lib/python3.7/site-packages/pandas/core/groupby/groupby.py", line 913, in <lambda>
    f = lambda x: func(x, *args, **kwargs)
  File "/app/tweets.py", line 125, in average_interp_named_timeseries
    t0 = timeseries.name.left
AttributeError: 'str' object has no attribute 'left'
```

**Review by beamalsky — COMMENTED** [review]
Very pleasing that we can `start_time`, thanks for doing that refactor! I'm getting an error—see comment below.

**Comment by beamalsky** [comment]
Fwiw I added a line to the README in [this PR](https://github.com/istheweatherweird/itww_bot/pull/35) with instructions how to use pytest locally: 

`docker-compose run --rm app pytest`

**Comment by potash** [comment]
Re: historical data it is on the hour but occasionally an hour is missing. My most recent push enforces coverage on historical data, too. If its inadequate it treats that year as missing. I ran a couple of tests and the biggest gap was 1 hour missing (so gap of 2 hours as opposed to 1).

Re: the error you're getting, weird! I don't get taht error. But I'm also not using the same versions of pandas etc. The groupby on my version conveniently puts the interval as the name attribute of the series. I think this was changed in pandas 1.0, looking into it now

**Comment by potash** [comment]
And gtk about docker, I will have to install it later to check. But when you run it at this time won't it try to set end_time to 6pm today and then give a coverage error?

**Comment by beamalsky** [comment]
Interesting! Part of the nice thing in Docker is that it enforces every environment using the same versions of packages. So on Heroku in production, it's also using everything listed in `requirements.txt`. We can change the line for `pandas` in that file to what you've been using, but also head off this kind of error by both using Docker in local dev. 

If I want to get results in local dev I've been adjusting the hour in `end_time`!

**Comment by beamalsky** [comment]
I merged master into here to capture the tests I added in #35

**Comment by potash** [comment]
Cool, yeah I need to invest a bit of time to setup docker (I don't even have it installed atm). While it should fix it to switch to pandas 0.25.x that seems wrong and I'd prefer to just get it working with pandas 1.0.1. But it's annoying, the earlier behavior made what I was doing much more straightforward!


 ---- On Mon, 09 Mar 2020 12:45:07 -0500 Bea Malsky <notifications@github.com> wrote ----

**Comment by beamalsky** [comment]
Whoops I didn't mean to add that README change to this branch, sorry! I think it's small enough though that we can just stick it in here.

**Comment by potash** [comment]
Ha, no problem. I'm merging this in!

Also, I turned logging output on in pytest. It is pretty useful for me to see what's going on: it prints the coverage and the temperatures etc. If you find it too much we can switch it back

**Comment by beamalsky** [comment]
I'm good with having pytest logging on!

**Pull request merged**